src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/DataFlowPass.vb

```diff
@@ -615,13 +615,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             ' If the state of the slot is 'assigned' we need to clear it 
             '    and possibly propagate it to all the parents
+            If Me._tryState IsNot Nothing Then
```

Comment:
I have refactored VB code roughly equivalent to C# here: https://github.com/dotnet/roslyn/pull/7728
That is more than enough for fixing the async analysis and tidying up.   
However it does not seem to work as intended in a case of more general "DataFlowsOut" analysis when struct partial assignments are involved (at least as I understand it should work). I will continue investigating that. If it is indeed broken, the reasons would not be in this code and the fix would be outside of the scope of this bug.